### PR TITLE
apply service charges

### PR DESCRIPTION
**Pull request by sapanakothari**

### CODE REVIEW

File: fineract_client.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 const (
-	fineractHost     = "https://13.209.34.65:8443" //"https://demo.openmf.org"
+	fineractHost     = "https://10.10.8.47:8443" //"https://13.209.34.65:8443" //"https://demo.openmf.org"
```
Comment (poorva29) [inline fineract_client.go]: Please remove commented lines

File: loan.go [hunk 2]
```diff
@@ -70,6 +71,17 @@ type GetLoanProductResponse struct {
 	InstallmentsCount  float64      `json:"numberOfRepayments"`
 	Type               InterestType `json:"interestType"`
 	InterestRate       float64      `json:"interestRatePerPeriod"`
+	Charge             []Charge     `json:"charges"`
+}
+
+type Charge struct {
+	Id     uint32  `json:"id"`
```
Comment (poorva29) [inline loan.go]: If there is absolute no code difference between `Charge` and `LoanCharge` we can reuse them

File: loan.go [hunk 3]
```diff
@@ -8,29 +8,30 @@ import (
...
+	InterestType                          string       `json:"interestType"`
+	InterestCalculationPeriodType         uint64       `json:"interestCalculationPeriodType"`
+	TransactionProcessingStrategy         uint64       `json:"transactionProcessingStrategyId"`
+	ExpectedDisbursementDate              string       `json:"expectedDisbursementDate"`
+	SubmittedOnDate                       string       `json:"submittedOnDate"`
+	LinkAccountId                         string       `json:"linkAccountId"`
+	RepaymentsStartingFromDate            string       `json:"repaymentsStartingFromDate,omitempty"`
+	InterestChargedFromDate               string       `json:"interestChargedFromDate,omitempty"`
+	AllowPartialPeriodInterestCalculation bool         `json:"allowPartialPeriodInterestCalcualtion,omitempty"`
+	Charge                                []LoanCharge `json:"charges"`
```
Comment (poorva29) [inline loan.go]: You can make variable name `Charges` as well

File: loan.go [hunk 4]
```diff
@@ -173,7 +186,7 @@ type LoanPeriod struct {
 	InterestOriginalDue             float64  `json:"interestOriginalDue,omitempty"`
 	InterestDue                     float64  `json:"interestDue,omitempty"`
 	InterestOutstanding             float64  `json:"interestOutstanding,omitempty"`
-	FeeChargesDue                   float64  `json:"feeChargesDue,omitempty"`
+	FeeCharges                      float64  `json:"feeChargesDue,omitempty"`
```
Comment (poorva29) [inline loan.go]: The variable name should be what it returns , please change this to `FeeChargesDue`

File: loan_test.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		t.Fatalf("Cannot get loan products: %v", err)
+	}
+
+	LoanSuite(t, client, fmt.Sprintf("%v", resp.LoanProducts[0].Id))
+}
+
+func LoanSuite(t *testing.T, client *Client, loanProductId string) {
+
+	t.Run("TestGetLoanProduct", func(t *testing.T) {
+		_, err := client.GetLoanProduct(loanProductId, &GetLoanProductRequest{})
```
Comment (poorva29) [inline loan_test.go]: A more accurate test would be where you check , loanProduct attributes to validate

File: loan.go (hunk 2 above)
Comment (sapanakothari) [reply loan.go]: there is a minor difference - LoanCharge has chargeId and Charge has id (json field names)

**Pull request merged**